### PR TITLE
Configurable MarketOnCloseOrder Buffer

### LINKED ISSUE #5481: MarketOnClose DefaultSubmissionTimeBuffer is Fixed
#### Expected Behavior

The DefaultSubmissionTimeBuffer of MarketOnClose orders is configurable. 
[This related thread](https://www.quantconnect.com/forum/discussion/11140) shows
> References:
> 
> NYSE
> https://www.nyse.com/publicdocs/nyse/markets/nyse/NYSE_Opening_and_Closing_Auctions_Fact_Sheet.pdf
> Market on Close (MOC)
> Can be entered until 3:50 pm
> 
> NASDAQ
> https://www.nasdaqtrader.com/content/productsservices/trading/crosses/openclose_faqs.pdf
> Market-on-close (MOC) orders must be received prior to 3:55 p.m

#### Actual Behavior

> MarketOnClose orders must be placed with at least a 16 minute buffer before market close.

#### Potential Solution

Make the [DefaultSubmissionTimeBuffer](https://github.com/QuantConnect/Lean/blob/master/Common/Orders/MarketOnCloseOrder.cs#L35) configurable.

#### Reproducing the Problem

Run [this backtest](https://www.quantconnect.com/terminal/processCache?request=embedded_backtest_9fe33b10a6670ea0a97fb91ac3c0fa84.html).

#### System Information

QC Cloud

#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by C-SELLERS**
#### Description

Add a public static TimeSpan var `SubmissionTimeBuffer` to `MarketOnCloseOrder` that is modifiable in special cases. 

#### Related Issue

Closes #5481

#### Motivation and Context

Allows more advanced users to modify this value for exchanges they know accepts OnClose Market Orders later than 15 minutes before close. 

#### Requires Documentation Change

Adding a note to the MarketOrderOnClose documentation would be nice for this.

#### How Has This Been Tested?

Tested using a simple algo to verify behavior.

```
    public class TestAlgo : QCAlgorithm
    {
        private bool Ordered;

        public override void Initialize()
        {
            SetStartDate(2013, 10, 4); //Set Start Date
            SetEndDate(2013, 10, 10); //Set End Date

            var ticker = "SPY";
            AddEquity(ticker, Resolution.Minute);
            Orders.MarketOnCloseOrder.SubmissionTimeBuffer = TimeSpan.FromMinutes(10);
        }

        public override void OnData(Slice slice)
        {
            if (!Ordered && Time.Hour == 15 && Time.Minute == 49)
            {
               // Will not throw an order error
                MarketOnCloseOrder("SPY", 2);
                Ordered = true;
            }

            if (Time.Hour == 15 && Time.Minute == 51)
            {
                // Will throw an order error
                MarketOnCloseOrder("SPY", 2);
            }
        }
    }
```

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Algorithm/QCAlgorithm.Trading.cs [hunk 1]
```diff
@@ -847,15 +847,16 @@ private OrderResponse PreOrderChecksImpl(SubmitOrderRequest request)
...
                 {
-                    // tell the user we require a 16 minute buffer, on minute data in live a user will receive the 3:44->3:45 bar at 3:45,
-                    // this is already too late to submit one of these orders, so make the user do it at the 3:43->3:44 bar so it's submitted
-                    // to the brokerage before 3:45.
+                    // Tell user the required buffer on these orders, also inform them it can be changed for special cases.
+                    // Default buffer is 15.5 minutes because with minute data a user will receive the 3:44->3:45 bar at 3:45,
+                    // if the latest time is 3:45 it is already too late to submit one of these orders
                     return OrderResponse.Error(request, OrderResponseErrorCode.MarketOnCloseOrderTooLate,
-                        "MarketOnClose orders must be placed with at least a 16 minute buffer before market close."
+                        $"MarketOnClose orders must be placed within {Orders.MarketOnCloseOrder.SubmissionTimeBuffer} before market close." +
```
Comment (Martin-Molinero) [inline Algorithm/QCAlgorithm.Trading.cs:858]: Cool! Let's add some regression algorithms doing so and asserting the behavior 👍

**Review by Martin-Molinero — APPROVED** [review]

File: Algorithm/QCAlgorithm.Trading.cs (hunk 1 above)
Comment (C-SELLERS) [reply Algorithm/QCAlgorithm.Trading.cs:858]: Gotcha 👍🏽

**Review by C-SELLERS — COMMENTED** [review]
Complete! 

 Regressions are in and verify behavior 👌🏽

**Review by Martin-Molinero — APPROVED** [review]
Nice! 🍬

**Pull request merged**